bin/scpipe/lib/sc/pipe.rb

```diff
@@ -69,37 +71,32 @@ def prepare_pipe
       end
 
       def run_pipe
-        @@pipeproc = Proc.new {
-          trap("INT") do
-            Process.exit
-          end
-          rundir = Dir.pwd
+        rundir = Dir.pwd
+        while @done==false do
           IO.popen("cd #{rundir} && #{SC.sclang_path.chomp} -d #{rundir.chomp} -i scvim", "w") do |sclang|
-            loop {
-              x = `cat #{@@pipe_loc}`
-              sclang.print x if x
-            }
+            @f = File.open(@@pipe_loc, "r")
+            begin
+              while x = @f.read do 
+                sclang.print x if x
+              end
+            rescue IOError => e
+            end
           end
-        }
-        $p = Process.fork { @@pipeproc.call }
+        end
       end
 
       def clean_up
         #if we get a hup then we kill the pipe process and
         #restart it
         trap("HUP") do
-          Process.kill("INT", $p)
-          $p = Process.fork { @@pipeproc.call }
+          @f.close
         end
 
         #clean up after us
         trap("INT") do
-          Process.kill("INT", $p)
-          remove_files
```

Comment:
Since `exit` is no longer called, this has been moved to after the `serve` method, where control flow passes after receiving an `INT` signal.